### PR TITLE
feat: if cardStyle has flexed specified, set the same value on PointerEventsView

### PR DESCRIPTION
**Pull request by tchayen**
### Problem

I was trying to achieve the following effect:

The transition used to dismiss modal by dragging it down, should only respond to gesture on the modal itself, not any background.

I found out that I can achieve it by using the way stack hierarchy works:
The screen component is wrapped by 'card' component which is wrapped in a pan gesture handler which is then wrapped in 'container' component.

### Obstacles

If I specified `height` and `width` for the card component, it worked perfectly. However, there was still a problem with dynamically sized modals, since `flex: 1` is present in container, card and another internal component wrapping the screen which is called `<PointerEventsView>`.

### Solution

The least invasive way to solve it which allows for dynamically sized modals and doesn't require API changes would be to copy the `flex: 1` style from `card` to `<PointerEventsView>`.

### CODE REVIEW

**Comment by tchayen** [comment]
Snapshots need update to match array with one more `undefined`.

File: src/views/Stack/Card.tsx [hunk 1]
```diff
@@ -638,6 +638,11 @@ export default class Card extends React.Component<Props> {
         : this.handleGestureEventHorizontal
       : undefined;
 
+    const overrideFlex =
+      cardStyle && cardStyle.hasOwnProperty('flex')
```
Comment (satya164) [inline src/views/Stack/Card.tsx]: This won't work since `style` prop can be an array, `null`, `undefined`, `false` etc.

Comment (tchayen) [reply src/views/Stack/Card.tsx]: Oops. Will fix.

Comment (tchayen) [reply src/views/Stack/Card.tsx]: I've changed the code to recognize and handle array. It automatically takes the last defined `flex` as it overwrites them when progressing, which is I believe the correct way handle that.

Values that falsify were already skipped and it stays that way.

**Comment by tchayen** [comment]
@satya164 Pointer events are responsible for something else if I understand correctly.

The point of this change was to be as little invasive as possible. In current state it doesn't change anything in the API, just tweaks edge case behaviour to allow users to gain more control via style interpolators.

**Comment by satya164** [comment]
Can you fix the unit tests? Then we can merge
